src/Tests/BusinessLayer/TelemetryReportingTest.cpp

```diff
@@ -0,0 +1,959 @@
+
+
+class TelemetryReportingTest : public ::testing::Test
+{
+
+public:
+    const unsigned int CHECK_SUM_LENGTH = 2;
+    const unsigned int COBS_ADDITIONAL_FRAME_DATA_SIZE = 2;
+
+    const unsigned int EXPECTED_PACKAGE_LENGTH_SEND_KEY_MOTOR = 47;
+    const unsigned int EXPECTED_PACKAGE_LENGTH_SEND_MOTOR_DETAILS = 73;
+    const unsigned int EXPECTED_PACKAGE_LENGTH_SEND_DRIVER_CONTROLS = 13;
+    const unsigned int EXPECTED_PACKAGE_LENGTH_SEND_MOTOR_FAULTS = 13;
+    const unsigned int EXPECTED_PACKAGE_LENGTH_SEND_BATTERY_FAULTS = 7;
+    const unsigned int EXPECTED_PACKAGE_LENGTH_SEND_BATTERY = 64;
+    const unsigned int EXPECTED_PACKAGE_LENGTH_SEND_CMU = 54;
+    const unsigned int EXPECTED_PACKAGE_LENGTH_SEND_MPPT = 14;
+    const unsigned int EXPECTED_PACKAGE_LENGTH_SEND_LIGHTS = 6;
+
+protected:
+
+    QScopedPointer<MockCommunicationService> communicationService_;
+    QScopedPointer<KeyMotorData> keyMotorData_;
+    QScopedPointer<MotorDetailsData> motor0DetailsData_;
+    QScopedPointer<MotorDetailsData> motor1DetailsData_;
+    QScopedPointer<DriverControlsData> driverControlsData_;
+    QScopedPointer<MotorFaultsData> motorFaultsData_;
+    QScopedPointer<BatteryFaultsData> batteryFaultsData_;
+    QScopedPointer<BatteryData> batteryData_;
+    QScopedPointer<CmuData> cmuData_;
+    QScopedPointer<MpptData> mpptData_;
+    QScopedPointer<LightsData> lightsData_;
+    QScopedPointer<View> view;
+
+    QScopedPointer<TelemetryReporting> telemetryReporting_;
+
+    virtual void SetUp()
+    {
+        communicationService_.reset(new MockCommunicationService());
+        keyMotorData_.reset(new KeyMotorData());
+        motor0DetailsData_.reset(new MotorDetailsData());
+        motor1DetailsData_.reset(new MotorDetailsData());
+        driverControlsData_.reset(new DriverControlsData());
+        motorFaultsData_.reset(new MotorFaultsData());
+        batteryFaultsData_.reset(new BatteryFaultsData());
+        batteryData_.reset(new BatteryData());
+        cmuData_.reset(new CmuData());
+        mpptData_.reset(new MpptData());
+        lightsData_.reset(new LightsData());
+        view.reset(new View());
+        telemetryReporting_.reset(new TelemetryReporting(*communicationService_,
+                                  *keyMotorData_,
+                                  *motor0DetailsData_,
+                                  *motor1DetailsData_,
+                                  *driverControlsData_,
+                                  *motorFaultsData_,
+                                  *batteryFaultsData_,
+                                  *batteryData_,
+                                  *cmuData_,
+                                  *mpptData_,
+                                  *lightsData_,
+                                  *view
+                                                        ));
+    }
+
+    void assertThatCOBScorrect(unsigned char* input, unsigned char* expected, const unsigned int inputLength)
```

Comment:
I changed it to expectCobsCorrect as I changed the use of ASSERT_THAT within the method to EXPECT_THAT